client/flutter/lib/pages/onboarding/onboarding_page.dart

```diff
@@ -19,12 +21,33 @@ class _OnboardingPageState extends State<OnboardingPage> {
   final FirebaseMessaging _firebaseMessaging = FirebaseMessaging();
   final PageController _pageController = PageController();
 
+  Map<String, IsoCountry> _countries;
+  bool _couldLoadCountries;
+  IsoCountry _selectedCountry;
+  bool _showCountryListPage = true;
+
   @override
   void dispose() {
     _pageController.dispose();
     super.dispose();
   }
 
+  @override
+  void initState() {
+    super.initState();
+    setupCountries();
+  }
+
+  void setupCountries() async {
+    _couldLoadCountries = await IsoCountryList().loadCountries();
+    if (_couldLoadCountries) {
+      _countries = IsoCountryList().countries;
+      final currentCountryCode = Localizations.localeOf(context).countryCode;
```

Comment:
It feels strange accessing the context from initState, but this works and [Flutter docs](https://api.flutter.dev/flutter/widgets/State/context.html) say context is created before initState is called, but I also see [SO answers](https://stackoverflow.com/a/49458289/2434476) [disagreeing](https://stackoverflow.com/questions/51766977/flutter-showdialogcontext-in-initstate-method#comment90496359_51766977) on this. Also, I wonder if it could cause problems to cause `setState` before the widget renders, which is possible here. This is working so far, but curious if @patniemeyer, @theswerd, or @britannio have any concerns or better approaches to this.

Reply:
instead of calling it directly in `initState` you can wrap it in `WidgetsBinding.instance.addPostFrameCallback((_) {...})` and put that in initState. It'll be called once after the first frame.

Also, since that's asynchronous I'd make sure that `mounted` is true as the widget could be disposed of before the asynchronous code completes.

Reply:
@britannio updated if you want to give it another look

Reply:
perfect